src/libcore/ptr/non_null.rs

```diff
@@ -69,6 +67,8 @@ impl<T: Sized> NonNull<T> {
     #[rustc_const_stable(feature = "const_nonnull_dangling", since = "1.32.0")]
     #[inline]
     pub const fn dangling() -> Self {
+        // SAFETY: mem::align_of() returns a valid, non-null pointer. The
```

Comment:
I'm not sure it would be right to call the pointer returned by `mem::align_of` "valid". It is still dangling.

Reply:
That's a good point. I thought of "valid" as in "non-null" and "usable", but it is in fact dangling and dangerous. We could simply put non-null and remove the valid to avoid confusion
```suggestion
        // SAFETY: mem::align_of() returns a non-null pointer. The
```

Reply:
Thanks a lot for all your comments by the way @LeSeulArtichaut

Reply:
No problem! I've got some spare time, being locked down at home :P

Reply:
`mem::align_of` doesn't return a pointer, it returns a `usize`. By casting it to a pointer, you are creating the lowest non-zero `*mut T` that is properly aligned.

Reply:
@ecstatic-morse Changed in 75a73d1